docs/migration-guide/to-17.md

```diff
@@ -123,6 +138,120 @@ root.walkRules((ruleNode) => {
 });
 ```
 
+### Changed `selector-max-*` syntax rules for standard CSS nesting and modern functional pseudo-classes
+
+We've changed the `selector` and `specificity` rules to account for CSS nesting and modern functional pseudo-classes, such as `:is()`, `:where()` and `:has()`.
+
+We've changed the following syntax rules:
+
+- `selector-max-attribute`
+- `selector-max-class`
+- `selector-max-combinators`
+- `selector-max-compound-selectors`
+- `selector-max-id`
+- `selector-max-pseudo-class`
+- `selector-max-type`
+- `selector-max-universal`
+
+So that they no longer:
+
+- desugar nesting
+- evaluate functional pseudo-classes separately, such as `:is()` and `:not()`
+
+Standard nesting often desugars to `:is()`. For example, the following:
+
+<!-- prettier-ignore -->
+```css
+#foo,
+.bar {
+  & a {}
+}
+```
+
+Is desugared to:
+
+<!-- prettier-ignore -->
+```css
+:is(#foo, .bar) a {}
+```
+
+This behaviour, when hidden, can lead to surprising lint problems being flagged. As we no longer desugar nesting in these rules, the selectors you see in your source files are now linted as-is, so there are no surprises.
+
+As these rules no longer evaluate functional pseudo-classes separately, you may need to increase your maxes. For example:
+
+<!-- prettier-ignore -->
+```css
+.foo:has(.bar) {}
+```
+
+```diff json
+-{ "selector-max-class": 1 }
++{ "selector-max-class": 2 }
+```
+
+### Changed `*-specificity` semantic rules for standard CSS nesting
+
+We've changed the following semantic rules to adhere to the [CSS Nesting specification](https://drafts.csswg.org/css-nesting/) when calculating specificity:
+
+- `no-descending-specificity`
+- `selector-max-specificity`
+
+The [specificity](https://drafts.csswg.org/selectors-4/#specificity-rules) of the [nesting selector](https://drafts.csswg.org/css-nesting/#nesting-selector) (`&`) is equal to the largest specificity among the complex selectors in the parent's selector list (identical to the behavior of [:is()](https://drafts.csswg.org/selectors-4/#matches-pseudo)).
```

Comment:
[suggestion] Minor. Please ignore them if you have any reasons.

- Links to the headings seem to be better than the specific words
- Wrap `:is()` with backticks for consistency

```suggestion
The [specificity](https://drafts.csswg.org/selectors-4/#specificity-rules) of the [nesting selector](https://drafts.csswg.org/css-nesting/#nest-selector) (`&`) is equal to the largest specificity among the complex selectors in the parent's selector list (identical to the behavior of [`:is()`](https://drafts.csswg.org/selectors-4/#matches)).
```